### ISSUE TITLE
RBF of CPFP tx

### ISSUE THREAD

**Issue by ziggamon** [spesmilo/electrum#6167]
Current master

If a big inbound transaction gets stuck due to high mempool it's common to do CPFP. However, if the initial CPFP doesn't work electrum does not allow the user to bump further.
CPFP option gets disabled on the original transaction (not quite sure why), and RBF of the child transaction shows an interface that measures tx fee in sat/B for the child transaction, requiring quite complex math to recalculate to what that should be based on the original transaction, the size of the child transaction.

This could be solved either by:
- Don't disable CPFP on a transaction after it has been CPFP'd.
- Show a full RBF UI to RBF the bump transaction, calculating the fees based on the entirety of the transactions that are being pulled.

**Comment by ecdsa** [comment]
A transaction cannot be CPFP'd if all its outputs that belong to you are already spent.
That is typically be the case if you already CPFP'd it.

**Comment by ziggamon** [comment]
Why not?

And if it really can't, then why not show the proper RBF UI? I'd say it's a common use case to try to RBF gradually and then increase it further if it's needed.

**Comment by ecdsa** [comment]
> Why not?
because that does not make sense. 
CPFP means that it shhould be possible to create a child transaction.

bumping the fee of the child tx is what you actually want to do.

**Comment by ziggamon** [comment]
>because that does not make sense.

I think you're confusing. CPFP happens to an incoming transaction. The child transaction is still incoming to my wallet. I could CPFP it again. 

I agree with you that bumping the fee of it would also help, and I'm asking for better UI for it. It's not reasonable that the user should have to calculate the fee manually and retrofit it into a sats/byte rate of the child transaction. It's far too complex. Better just show the original CPFP dialog and let the user set the fee for the entire block of transactions that are being confirmed.

**Comment by ecdsa** [comment]
> I think you're confusing. CPFP happens to an incoming transaction. The child transaction is still incoming to my wallet. I could CPFP it again

So you want to create a grand-child transaction?
That sounds pretty wasteful compared to RBF
However, I do not think that is disabled.

**Comment by ziggamon** [comment]
It is wasteful, a decent RBF UI for CPFP would also work of course, as mentioned.

>However, I do not think that is disabled.

It is.

Please try this out yourself by CPFPing a transaction, and then try to bump it further and you'll see what I mean.

**Comment by ecdsa** [comment]
oh indeed, I just checked the code, CPFP is disabled when RBF is possible.
which makes sense, we should not propose CPFP when RBF is possible.

**Comment by ziggamon** [comment]
Yes, and now please check what the RBF feature looks like when RBFing a CPFP transaction.

**Comment by SomberNight** [comment]
> Show a full RBF UI to RBF the bump transaction, calculating the fees based on the entirety of the transactions that are being pulled.

RIght, so when RBF fee-bumping a tx, we could consider its unconfirmed ancestors too.
I think the proposed feature makes sense in spirit but I am not sure how it should work exactly.
That is, how many ancestors do we go up?

I guess what would be reasonable is to either
- only consider at most one ancestor (and only if already in the wallet history),
- or alternatively all ancestors that are already in the wallet history.

It is not reasonable imho to retrieve ancestors that are not already in the wallet history.

Sort of related: https://github.com/spesmilo/electrum/issues/3470

**Comment by ziggamon** [comment]
Right, so maybe leave the #3470 conversation for that ticket (I would argue that all unconfirmed ancestors should absolutely be counted, otherwise UX gets weird), but that's an entirely separate issue. For now I'd like RBF of CPFP tx's to work in the expected manner, showing UI that weighs the whole transaction, and possibly also be reachable in the right-click menu from the original transaction.

**Comment by farmerwampum** [comment]
>     * or alternatively all ancestors that are already in the wallet history.

This is the behavior I expected
